plugins/jobs/src/js/JobsRunNow.js

```diff
@@ -21,16 +19,11 @@ const runNowJobGraphql = id => {
   return graphqlObservable(runNowJobMutation, defaultSchema, id);
 };
 
-export const runNowEvent$ = new Subject();
-export const jobsRunNow$ = runNowEvent$.switchMap(id => {
-  return runNowJobGraphql(id);
-});
-
-export const jobsRunNowAction = id => {
+export default function(id) {
   return {
     label: "Run Now",
     onItemSelect() {
-      runNowEvent$.next({ id });
+      runNowJobGraphql({ id }).subscribe();
```

Comment:
Don't really like that I have to add the subscribe to run it. But the previous way where is create a subject and then merge the 2 to run .next() looks like a bit hacky as well.

Reply:
Calling subscribe, means you have to do:
```js
const subscription = stream$.subscribe();

//somehwere later.
subscription.unsubscribe();
```
Otherwise, you may be causing a memory leak.

The merging of the subject is actually the correct way to go. That is actually a pattern you might hope to find on any use of the subject as a signal/event passed up. You can look at `createEventHandler` from recompose (_https://github.com/acdlite/recompose/blob/master/docs/API.md#createeventhandler). The subject merged/combined/filter with other operators, is the way to treat the event (stream), and the calling of next is the handler (the function to pass the signal up).